awx/ui/client/src/templates/workflows/add-workflow/workflow-add.controller.js

```diff
@@ -52,6 +52,138 @@ export default [
          $scope.surveyTooltip = i18n._('Please save before adding a survey to this workflow.');
 
+        $scope.webhook_service_options = null;
+
+        // populate webhook service choices
+        GetChoices({
+            scope: $scope,
+            url: GetBasePath('workflow_job_templates'),
+            field: 'webhook_service',
+            variable: 'webhook_service_options',
+        });
+
+        // set initial val for webhook checkbox
+        $scope.enable_webhook = false;
+
+        //
+        // webhook credential - all handlers, dynamic state, etc. live here
+        //
+
+        $scope.webhookCredential = {
+            id: null,
+            name: null,
+            isModalOpen: false,
+            isModalReady: false,
+            modalSelectedId: null,
+            modalSelectedName: null,
+            modalBaseParams: {
+                order_by: 'name',
+                page_size: 5,
+                credential_type__namespace: null,
+            },
+            modalTitle: i18n._('Select Webhook Credential'),
+        };
+
+        $scope.handleWebhookCredentialLookupClick = () => {
+            $scope.webhookCredential.modalSelectedId = $scope.webhookCredential.id;
+            $scope.webhookCredential.isModalOpen = true;
+        };
+
+        $scope.handleWebhookCredentialTagDelete = () => {
+            $scope.webhookCredential.id = null;
+            $scope.webhookCredential.name = null;
+        };
+
+        $scope.handleWebhookCredentialModalClose = () => {
+            $scope.webhookCredential.isModalOpen = false;
+            $scope.webhookCredential.isModalReady = false;
+        };
+
+        $scope.handleWebhookCredentialModalReady = () => {
+            $scope.webhookCredential.isModalReady = true;
+        };
+
+        $scope.handleWebhookCredentialModalItemSelect = (item) => {
+            $scope.webhookCredential.modalSelectedId = item.id;
+            $scope.webhookCredential.modalSelectedName = item.name;
+        };
+
+        $scope.handleWebhookCredentialModalCancel = () => {
+            $scope.webhookCredential.isModalOpen = false;
+            $scope.webhookCredential.isModalReady = false;
+            $scope.webhookCredential.modalSelectedId = null;
+            $scope.webhookCredential.modalSelectedName = null;
+        };
+
+        $scope.handleWebhookCredentialSelect = () => {
+            $scope.webhookCredential.isModalOpen = false;
+            $scope.webhookCredential.isModalReady = false;
+            $scope.webhookCredential.id = $scope.webhookCredential.modalSelectedId;
+            $scope.webhookCredential.name = $scope.webhookCredential.modalSelectedName;
+            $scope.webhookCredential.modalSelectedId = null;
+            $scope.webhookCredential.modalSelectedName = null;
+        };
+
+        $scope.handleWebhookKeyButtonClick = () => {};
+
+        $('#content-container').append($compile(`
```

Comment:
same thing...is there a way to not use $compile?

Reply:
see comment above: https://github.com/ansible/awx/pull/4436#discussion_r329784135